kroxylicious-filters/kroxylicious-encryption/src/main/java/io/kroxylicious/filter/encryption/ResilientKms.java

```diff
@@ -39,10 +40,13 @@ public class ResilientKms<K, E> implements Kms<K, E> {
     private final BackoffStrategy strategy;
     private final int retries;
 
-    private ResilientKms(Kms<K, E> inner,
-                         ScheduledExecutorService executorService,
-                         BackoffStrategy backoffStrategy,
+    private ResilientKms(@NonNull Kms<K, E> inner,
+                         @NonNull ScheduledExecutorService executorService,
+                         @NonNull BackoffStrategy backoffStrategy,
                          int retries) {
+        Objects.requireNonNull(inner);
+        Objects.requireNonNull(executorService);
+        Objects.requireNonNull(backoffStrategy);
         this.inner = inner;
         this.executorService = executorService;
         strategy = backoffStrategy;
```

Comment:
Nit: Personally I prefer
```suggestion
        this.inner = Objects.requireNonNull(inner);
        this.executorService = Objects.requireNonNull(executorService);
        strategy = Objects.requireNonNull(backoffStrategy);
```